kitsune/search/v2/management/commands/es7_init.py

```diff
@@ -29,11 +35,32 @@ def handle(self, *args, **kwargs):
         if limit:
             doc_types = [dt for dt in doc_types if dt.__name__ in limit]
 
+        timestamp = datetime.now(tz=timezone.utc)
+
         for dt in doc_types:
-            self.stdout.write("Initializing: {}".format(dt.__name__))
-            if kwargs["delete"]:
-                try:
-                    dt._index.delete()
-                except NotFoundError:
-                    pass
-            dt.init()
+            print(f"Initializing: {dt.__name__}")
+
+            migrate_writes = kwargs["migrate_writes"]
+            migrate_reads = kwargs["migrate_reads"]
+
+            if not (migrate_reads or migrate_writes):
+                index = dt.write_alias_points_at
+                if not index:
+                    print("First time running, creating index and aliases:")
+                    migrate_writes = True
+                    migrate_reads = True
+                else:
+                    print("Updating index")
+                    dt.init(index=index)
+
+            try:
```

Comment:
I would wrap each operation in their own `try/except` clause

Reply:
I wasn't sure if we'd want to migrate reads if writes failed for whatever reason, that said I'm not sure we'll be migrating both at once in any situation other than local dev.